sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -1111,6 +1134,27 @@ public virtual async ValueTask DisposeAsync()
                         eventData.ClearPublishingState();
                     }
 
+                    // Reset the partition state and options to ensure that future attempts
+                    // are safe and do not risk data loss by reusing the same producer group identifier.
+
+                    if (!Options.PartitionOptions.TryGetValue(options.PartitionId, out var partitionOptions))
```

Comment:
Does this need any locking?

Reply:
Yup; we acquire a semaphore on L1090 that covers the entire publishing operation.  There's only one active `SendAsync` call allowed with the idempotent option, as the entire thing is stateful.

Reply:
Note to self:  Update the pattern there; we can probably avoid the async call in most cases.